LiteCore/RevTrees/RevTree.hh

```diff
@@ -26,36 +25,36 @@ namespace litecore {
     /** In-memory representation of a single revision's metadata. */
     class Rev {
       public:
-        const RevTree* owner;
-        const Rev*     parent;
-        revid          revID;    /**< Revision ID (compressed) */
-        sequence_t     sequence; /**< DB sequence number that this revision has/had */
+        const RevTree* owner{};
+        const Rev*     parent{};
+        revid          revID;      /**< Revision ID (compressed) */
+        sequence_t     sequence{}; /**< DB sequence number that this revision has/had */
 
-        slice body() const;
+        [[nodiscard]] slice body() const;
 
-        bool isBodyAvailable() const FLPURE { return _body.buf != nullptr; }
+        [[nodiscard]] bool isBodyAvailable() const FLPURE { return _body.buf != nullptr; }
 
-        bool isLeaf() const FLPURE { return (flags & kLeaf) != 0; }
+        [[nodiscard]] bool isLeaf() const FLPURE { return (flags & kLeaf) != 0; }
 
-        bool isDeleted() const FLPURE { return (flags & kDeleted) != 0; }
+        [[nodiscard]] bool isDeleted() const FLPURE { return (flags & kDeleted) != 0; }
 
-        bool hasAttachments() const FLPURE { return (flags & kHasAttachments) != 0; }
+        [[nodiscard]] bool hasAttachments() const FLPURE { return (flags & kHasAttachments) != 0; }
 
-        bool isNew() const FLPURE { return (flags & kNew) != 0; }
+        [[nodiscard]] bool isNew() const FLPURE { return (flags & kNew) != 0; }
 
-        bool isConflict() const FLPURE { return (flags & kIsConflict) != 0; }
+        [[nodiscard]] bool isConflict() const FLPURE { return (flags & kIsConflict) != 0; }
 
-        bool isClosed() const FLPURE { return (flags & kClosed) != 0; }
+        [[nodiscard]] bool isClosed() const FLPURE { return (flags & kClosed) != 0; }
 
-        bool keepBody() const FLPURE { return (flags & kKeepBody) != 0; }
+        [[nodiscard]] bool keepBody() const FLPURE { return (flags & kKeepBody) != 0; }
 
-        bool isActive() const FLPURE;
+        [[nodiscard]] bool isActive() const FLPURE;
```

Comment:
Can't we just make this the default somehow? 😝

Reply:
That is something I wonder myself. I'll do some research on it. At first I avoided the warning specifically because of these. I hate putting `[[nodiscard]]` next to virtually every bloody function.